pkg/logs/docker/docker.go

```diff
@@ -35,11 +51,61 @@ func ParseMessage(msg []byte) (string, string, []byte, error) {
 	}
 
 	// timestamp goes from byte 8 till first space
-	to := bytes.Index(msg[messageHeaderLength:], []byte{' '})
-	if to == -1 {
-		return "", "", nil, errors.New("Can't parse docker message: expected a whitespace after header")
+	headerLen := GetHeaderLength(msg)
+	if headerLen == 0 {
+		return Message{}, errors.New("Can't parse docker message: expected a whitespace after header")
+	}
+	timestamp := string(msg[dockerHeaderLength : headerLen-1])
+
+	return Message{
+		Content:   msg[headerLen:],
+		Status:    status,
+		Timestamp: timestamp,
+	}, nil
+}
+
+// removePartialHeaders removes the 8 byte header, timestamp, and space that occurs between 16Kb section of a log.
+// If a docker log is greater than 16Kb, each 16Kb partial section will have a header, timestamp, and space in front of it.
+// For example, a message that is 35kb will be of the form: `H M1H M2H M3` where "H" is what pre-pends each 16 Kb section.
+// This function removes the "H " between two partial messages sections while leaving the very first "H ".
+// Input:
+//   H M1H M2H M3
+// Output:
+//   H M1M2M3
+func removePartialHeaders(msgToClean []byte) []byte {
+	msg := []byte{}
+	headerLen := GetHeaderLength(msgToClean)
+	start := 0
+	end := min(len(msgToClean), dockerBufferSize+headerLen)
+
+	for end > 0 && headerLen > 0 {
+		msg = append(msg, msgToClean[start:end]...)
+		msgToClean = msgToClean[end:]
+		headerLen = GetHeaderLength(msgToClean)
+		start = headerLen
+		end = min(len(msgToClean), dockerBufferSize+headerLen)
+	}
+
+	return msg
+}
+
+// GetHeaderLength returns the length of the 8 bytes header, timestamp, and space
+// that is in front of each message.
+func GetHeaderLength(msg []byte) int {
```

Comment:
I think this name is confusing, as we have earlier `const dockerHeaderLength`. Could we find a better name for one or the other?

Reply:
looks good thanks